scrapy/core/downloader/tls.py

```diff
@@ -48,6 +48,11 @@ def _identityVerifyingInfoCallback(self, connection, where, ret):
                         'Remote certificate is not valid for hostname "{}"; {}'.format(
                             self._hostnameASCII, e))
 
+                except ValueError as e:
+                    logger.warning(
+                        'SSL/TLS verification failed for hostname "{}"; {}'.format(
+                            self._hostnameASCII, e))
```

Comment:
It's correct that the trouble comes from passing the IP address to verify_hostname.
[SubjectAltName apparently can include IP addresses](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Deployment_Guide/sssd-ldap-domain-ip.html) but we need to check what happens when remote certificate does include it, matching what the client connected to.
The error with scrapy 1.0.6 looks identical but the agent implementation made it "work".
This PR is definitely not pretty.
Would you suggest coming up with a proper verification implementation?